src/main/java/ca/wolfram/beta/symath/operations/AddNode.java

```diff
@@ -35,19 +35,49 @@ public NodeType getType() {
         return NodeType.ADD;
     }
 
+    /**
+     * Extracts all constant integers and appends it as one {@link ca.wolfram.beta.symath.base.ConstantNode} if the sum != 0
+     * Flattens children if they are also of type ADD
+     *
+     * @return true is isConstant, false otherwise
+     */
     @Override
     public boolean simplify() {
-        int constant = 0;
-        final Iterator<MathNode> each = getChildren().iterator();
+        int constant = simplify(0, 0);
+        if (constant != 0)
+            children.add(BaseNode.create(constant));
+        return super.simplify();
+    }
+
+    /**
+     * Helper method to help with simplification
+     * Starts the listIterator at index, with history of constant, and continues from there
+     *
+     * @param index    children index to start iteration
+     * @param constant integer summation of child nodes before index
+     * @return new integer summation
+     */
+    private int simplify(int index, int constant) {
+        List<MathNode> childrenToRaise = new ArrayList<>(); //children to be flattened to this node level
```

Comment:
for comments in C++, there should be 2 spaces between the code and // and 1 space before the message begins. not sure about java.